### PR TITLE
Add unit conversion component

### PR DESCRIPTION
**Pull request by Rashmil-1999**
run `npm i` to install the new package `convert-units` which provides a simple conversion of units. Added dropdowns on environment tabs but the state is in the Sidebar component so that the same metrics are preserved in the successive searches.

### CODE REVIEW

**Review by ka7eh — CHANGES_REQUESTED** [review]
The pieces look good and work fine, but I think we should update `convert-units` version as the current stable one is pretty old and looks like a new release is under way.
Also, see my other comment on managing the state of selected units.

File: package.json [hunk 1]
```diff
@@ -21,7 +21,9 @@
     "@emotion/styled": "^11.3.0",
     "@mui/lab": "^5.0.0-alpha.53",
     "@mui/material": "^5.0.6",
+    "@types/convert-units": "^2.3.5",
```
[the remark this answers is not in the document]
Comment (Rashmil-1999) [reply package.json:24]: Okay, then I will make the new changes and then update this PR.

File: src/components/Layouts/SinglePage.tsx [hunk 2]
```diff
@@ -83,6 +88,15 @@ const Layout: React.FC<Props> = ({ children }: Props) => {
...
                             Analysis
                         </Button>
+                        <Button
+                            variant="outlined"
+                            size="small"
+                            startIcon={<Icon baseClassName="icons">build</Icon>}
+                            onClick={openPreferencesDialog}
+                            sx={{ color: 'white' }}
+                        >
+                            Unit Preferences
```
Comment (ka7eh) [inline src/components/Layouts/SinglePage.tsx:98]: Let's rename this to "Preferences" and move it to the right of the API button. We might add more preferences to this later.

**Review by ka7eh — CHANGES_REQUESTED** [review]

File: src/store/reducers.ts [hunk 3]
```diff
@@ -34,6 +34,26 @@ export const dataReducers = (state: DataState, action: DataAction): DataState =>
                 ...state,
                 faoAreas: action.faoAreas
             };
+        case 'updateTempFromUnit':
```
[the remark this answers is not in the document]
Comment (Rashmil-1999) [reply src/store/reducers.ts:37]: Okay got it.

File: src/components/Layouts/SinglePage.tsx (hunk 2 above)
Comment (Rashmil-1999) [reply src/components/Layouts/SinglePage.tsx:98]: Yup will do that.

**Review by ka7eh — APPROVED** [review]

**Pull request merged**